pkg/simple/client/es/query/builder.go

```diff
@@ -460,10 +460,17 @@ func (m *Terms) IsValid() bool {
 
 func NewTerms(key string, val interface{}) *Terms {
 
-	if reflect.ValueOf(val).IsNil() {
+	if val == nil {
```

Comment:
The `terms` query only accepts slice parameters, so here it is only necessary to check if `val` is nil or an empty slice. The `switch` statement below can also be omitted.